src/main/java/org/dimdev/jeid/mixin/core/world/MixinStateImplementation.java

```diff
@@ -0,0 +1,36 @@
+package org.dimdev.jeid.mixin.core.world;
+
+import com.google.common.collect.*;
+import java.util.*;
+import net.minecraft.block.properties.*;
+import net.minecraft.block.state.BlockStateContainer.*;
+import org.spongepowered.asm.mixin.*;
+import org.spongepowered.asm.mixin.injection.*;
+import org.spongepowered.asm.mixin.injection.callback.*;
+
+@Mixin(StateImplementation.class)
+public class MixinStateImplementation {
+
+    private int hashCodeValue;
+
+    @Shadow
+    private ImmutableMap<IProperty<?>, Comparable<?>> properties;
+
+    @Inject(
+        method = "<init>",
+        at = @At("RETURN")
+    )
+    private void initHashCode(CallbackInfo ci) {
+        hashCodeValue = Objects.hash(this, properties);
```

Comment:
Using `this` in the hash is incorrect; this is a mixin so `this` would just call this mixin's `hashCode() -> hashCodeValue -> 0`. I think you meant to shadow `StateImplementation#block` and use that as part of the hash?